core/domain/prod_validation_jobs_one_off_test.py

```diff
@@ -2188,6 +2190,209 @@ def test_model_with_invalid_collection_rights_related_property(self):
+                'u"Entity id 3: based on field topic_ids having value topic, '
+                'expect model TopicModel with id topic but it doesn\'t exist"]]'
+            )]
+        run_job_and_check_output(
+            self, expected_output, sort=True, literal_eval=True)
+
+    def test_missing_exp_model_failure(self):
+        exp_model = exp_models.ExplorationModel.get_by_id('1')
+        exp_model.delete(feconf.SYSTEM_COMMITTER_ID, '', [])
+
+        expected_output = [(
+            u'[u\'failed validation check for exploration_ids field check '
+            'of ExplorationOpportunitySummaryModel\', '
+            '[u"Entity id 1: based on field exploration_ids having '
+            'value 1, expect model ExplorationModel with id 1 but it '
+            'doesn\'t exist"]]'), (
+                u'[u\'fully-validated ExplorationOpportunitySummaryModel\','
+                ' 2]')]
+        run_job_and_check_output(self, expected_output, sort=True)
+
+    def test_model_with_invalid_content_count(self):
+        self.model_instance_1.content_count = 10
+        self.model_instance_1.put()
+        expected_output = [
+            (
+                u'[u\'failed validation check for content count check '
+                'of ExplorationOpportunitySummaryModel\', '
+                '[u"Entity id 1: Content count: 10 does not match the '
+                'content count of external exploration model: 2"]]'
+            ), u'[u\'fully-validated ExplorationOpportunitySummaryModel\', 2]']
+        run_job_and_check_output(
+            self, expected_output, sort=True, literal_eval=True)
+
+    def test_model_with_invalid_translation_counts(self):
+        self.model_instance_1.translation_counts = {'hi': 2}
+        self.model_instance_1.put()
+        expected_output = [
+            (
+                u'[u\'failed validation check for translation counts check '
+                'of ExplorationOpportunitySummaryModel\', '
+                '[u"Entity id 1: Translation counts: {u\'hi\': 2} does not '
+                'match the translation counts of external exploration model: '
+                '{}"]]'
+            ), u'[u\'fully-validated ExplorationOpportunitySummaryModel\', 2]']
+        run_job_and_check_output(
+            self, expected_output, sort=True, literal_eval=True)
+
+    def test_model_with_invalid_chapter_title(self):
+        self.model_instance_1.chapter_title = 'Invalid title'
+        self.model_instance_1.put()
+        expected_output = [
+            (
+                u'[u\'failed validation check for chapter title check '
+                'of ExplorationOpportunitySummaryModel\', '
+                '[u"Entity id 1: Chapter title: Invalid title does not match '
+                'the chapter title of external story model: Node 1"]]'
+            ), u'[u\'fully-validated ExplorationOpportunitySummaryModel\', 2]']
+        run_job_and_check_output(
+            self, expected_output, sort=True, literal_eval=True)
+
```

Comment:
Please add a test for one of the properties in `_get_external_model_properties` function as well.

Reply:
`_get_external_model_properties` is a private method, so I have added tests which correspond to all the external modal properties. You can check the test coverage is also 100%. Let me know if I'm missing something?

Reply:
I am sorry but I did not find the test - i.e. do you have a test for checking that `item.story_title` matches the title of story or `item.topic_name` matches name of the topic?

Reply:
isn't these test happens in the validate method and the validate method run on every test? Like I don't see any test for line 1198 in the existing test file (i.e,  collection tags matches collectionsummary model tags.) Do I have to add an explicit test for this one?

Reply:
I have actually added a test only for one of the properties mismatch in the dict of properties (You can check `test_model_with_invalid_collection_related_property` and `test_model_with_invalid_collection_rights_related_property`). So, similarly you can add a test here.

Reply:
Done!